packages/aragon-wrapper/src/index.js

```diff
@@ -1242,6 +1246,99 @@ export default class Aragon {
     return []
   }
 
+  /**
+   * Calculate the transaction path for a basket of intents.
+   * Expects the `intentBasket` to be an array of tuples holding the following:
+   *   {string}   destination: destination address
+   *   {string}   methodName: method to invoke on destination
+   *   {Array<*>} params: method params
+   * These are the same parameters as the ones used for `getTransactionPath()`
+   *
+   * Allows user to specify how many of the intents should be checked.
+   * `checkMode` supports:
+   *   'all': All intents will be checked to make sure they use the same forwarding path.
+   *   'single': assumes all intents can use the path found from the first intent
+   *
+   * @param  {Array<Array<string, string, Array<*>>>} intentBasket Intents
+   * @param  {Object} [options]
+   * @param  {string} [options.checkMode] Path checking mode
+   * @return {Promise<Array<Object>>} An array of Ethereum transactions that describe each step in the path
+   */
+  async getTransactionPathForIntentBasket (intentBasket, { checkMode = 'all' } = {}) {
+    // Get transaction paths for entire basket
+    const intentsToCheck =
+      checkMode === 'all'
+        ? intentBasket // all -- use all intents
+        : checkMode === 'single'
+          ? [intentBasket[0]] // single -- only use first intent
+          : []
+    const intentPaths = await Promise.all(
+      intentsToCheck.map(
+        ([destination, methodName, params]) =>
+          this.getTransactionPath(destination, methodName, params))
+    )
+
+    // Check paths all match
+    const individualPaths = intentPaths
```

Comment:
What do you think about breaking this into a separate utility function to check for path uniqueness?